### ISSUE TITLE
following quickstart runs into errors - latest poetry missing some dependencies, such as cleo.  

### ISSUE THREAD

**Issue by dlethin** [cjolowicz/cookiecutter-hypermodern-python#1096]
Full disclaimer -- I am *not* a seasoned programmer.  I came across this project trying to modernize my toolchain.

This reported issue is probably not a bug per-say -- just some feedback as I run into an error following the quickstart guide to setup and then trying to run nox on a repo I just setup.

Seems there's a dependency on 'cleo' missing when running poetry via nox:

```
❯ nox -v -r -s pre-commit
nox > Running session pre-commit
nox > Creating virtual environment (virtualenv) using python3.10 in .nox/pre-commit
created virtual environment CPython3.10.1.final.0-64 in 253ms
  creator CPython3Posix(dest=/Users/douglethin/bld/scratch/topper/.nox/pre-commit, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/Users/douglethin/Library/Application Support/virtualenv)
    added seed packages: pip==21.3.1, setuptools==59.2.0, wheel==0.37.0
  activators BashActivator,CShellActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator

nox > poetry export --format=requirements.txt --dev --without-hashes
Traceback (most recent call last):
  File "/Users/douglethin/.poetry/bin/poetry", line 25, in <module>
    from poetry.console import main
  File "/Users/douglethin/.poetry/lib/poetry/console/__init__.py", line 1, in <module>
    from .application import Application
  File "/Users/douglethin/.poetry/lib/poetry/console/application.py", line 3, in <module>
    from cleo import Application as BaseApplication
ModuleNotFoundError: No module named 'cleo'
nox > Command poetry export --format=requirements.txt --dev --without-hashes failed with exit code 1:
nox > Session pre-commit failed.
```

It seems the problem is related to this nox issue - [ModuleNotFoundError: No module named 'cleo'](https://github.com/python-poetry/poetry/issues/3345)

It seems one [comment](https://github.com/python-poetry/poetry/issues/553#issuecomment-994303497) suggested to install the following:

```
 pip install cleo tomlkit poetry.core requests cachecontrol cachy html5lib pkginfo virtualenv lockfile
```

There are a lot of virtual environments in play with this toolchain.   I'm having a hard time figuring out which tool to install these dependencies in.

the only way I got it to temporarily work was to install this into the code-coverage using the following command:

```
./.nox/coverage/bin/pip3 install cleo tomlkit poetry.core requests cachecontrol cachy html5lib pkginfo virtualenv lockfile
Collecting cleo
  Using cached cleo-0.8.1-py2.py3-none-any.whl (21 kB)
Collecting tomlkit
  Using cached tomlkit-0.8.0-py3-none-any.whl (33 kB)
Collecting poetry.core
  Using cached poetry_core-1.0.7-py2.py3-none-any.whl (424 kB)
Collecting requests
  Using cached requests-2.26.0-py2.py3-none-any.whl (62 kB)
Collecting cachecontrol
  Using cached CacheControl-0.12.10-py2.py3-none-any.whl (20 kB)
Collecting cachy
  Using cached cachy-0.3.0-py2.py3-none-any.whl (20 kB)
Collecting html5lib
  Using cached html5lib-1.1-py2.py3-none-any.whl (112 kB)
Collecting pkginfo
  Using cached pkginfo-1.8.2-py2.py3-none-any.whl (26 kB)
Collecting virtualenv
  Using cached virtualenv-20.10.0-py2.py3-none-any.whl (5.6 MB)
Collecting lockfile
  Using cached lockfile-0.12.2-py2.py3-none-any.whl (13 kB)
Collecting clikit<0.7.0,>=0.6.0
  Using cached clikit-0.6.2-py2.py3-none-any.whl (91 kB)
Collecting charset-normalizer~=2.0.0
  Using cached charset_normalizer-2.0.9-py3-none-any.whl (39 kB)
Collecting urllib3<1.27,>=1.21.1
  Using cached urllib3-1.26.7-py2.py3-none-any.whl (138 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2021.10.8-py2.py3-none-any.whl (149 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.3-py3-none-any.whl (61 kB)
Collecting msgpack>=0.5.2
  Using cached msgpack-1.0.3-cp310-cp310-macosx_10_9_x86_64.whl (75 kB)
Collecting six>=1.9
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting webencodings
  Using cached webencodings-0.5.1-py2.py3-none-any.whl (11 kB)
Collecting filelock<4,>=3.2
  Using cached filelock-3.4.0-py3-none-any.whl (9.8 kB)
Collecting platformdirs<3,>=2
  Using cached platformdirs-2.4.0-py3-none-any.whl (14 kB)
Collecting backports.entry-points-selectable>=1.0.4
  Using cached backports.entry_points_selectable-1.1.1-py2.py3-none-any.whl (6.2 kB)
Collecting distlib<1,>=0.3.1
  Using cached distlib-0.3.4-py2.py3-none-any.whl (461 kB)
Collecting pastel<0.3.0,>=0.2.0
  Using cached pastel-0.2.1-py2.py3-none-any.whl (6.0 kB)
Collecting pylev<2.0,>=1.3
  Using cached pylev-1.4.0-py2.py3-none-any.whl (6.1 kB)
Collecting crashtest<0.4.0,>=0.3.0
  Using cached crashtest-0.3.1-py3-none-any.whl (7.0 kB)
Installing collected packages: urllib3, pylev, pastel, idna, crashtest, charset-normalizer, certifi, webencodings, six, requests, platformdirs, msgpack, filelock, distlib, clikit, backports.entry-points-selectable, virtualenv, tomlkit, poetry.core, pkginfo, lockfile, html5lib, cleo, cachy, cachecontrol
Successfully installed backports.entry-points-selectable-1.1.1 cachecontrol-0.12.10 cachy-0.3.0 certifi-2021.10.8 charset-normalizer-2.0.9 cleo-0.8.1 clikit-0.6.2 crashtest-0.3.1 distlib-0.3.4 filelock-3.4.0 html5lib-1.1 idna-3.3 lockfile-0.12.2 msgpack-1.0.3 pastel-0.2.1 pkginfo-1.8.2 platformdirs-2.4.0 poetry.core-1.0.7 pylev-1.4.0 requests-2.26.0 six-1.16.0 tomlkit-0.8.0 urllib3-1.26.7 virtualenv-20.10.0 webencodings-0.5.1
```

Then running the pre-commit session worked:

```
nox > Re-using existing virtual environment at .nox/pre-commit.
nox > poetry export --format=requirements.txt --dev --without-hashes
alabaster==0.7.12; python_version >= "3.6"
aspy.refactor-imports==2.2.0; python_full_version >= "3.6.1"
atomicwrites==1.4.0; python_version >= "3.6" and python_full_version < "3.0.0" and sys_platform == "win32" or sys_platform == "win32" and python_version >= "3.6" and python_full_version >= "3.4.0"
attrs==21.2.0; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.5.0" and python_version >= "3.6"
babel==2.9.1; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.4.0" and python_version >= "3.6"
backports.entry-points-selectable==1.1.1; python_version >= "2.7" and python_full_version >= "3.6.1"
bandit==1.7.1; python_version >= "3.5"
beautifulsoup4==4.10.0; python_full_version > "3.0.0" and python_version >= "3.6"
black==21.11b1; python_full_version >= "3.6.2"
cached-property==1.5.2; python_full_version >= "3.6.1"
certifi==2021.10.8; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.6.0" and python_version >= "3.6"
cfgv==3.3.1; python_full_version >= "3.6.1"
charset-normalizer==2.0.8; python_full_version >= "3.6.0" and python_version >= "3.6"
click==8.0.3; python_version >= "3.6"
colorama==0.4.4; platform_system == "Windows" and python_version >= "3.6" and python_full_version >= "3.6.2" and sys_platform == "win32" and (python_version >= "3.6" and python_full_version < "3.0.0" and sys_platform == "win32" or sys_platform == "win32" and python_version >= "3.6" and python_full_version >= "3.5.0") and (python_version >= "3.5" and python_full_version < "3.0.0" and platform_system == "Windows" or platform_system == "Windows" and python_version >= "3.5" and python_full_version >= "3.5.0")
coverage==6.1.2; python_version >= "3.6"
darglint==1.8.1; python_version >= "3.6" and python_version < "4.0"
distlib==0.3.3; python_full_version >= "3.6.1"
docutils==0.17.1; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.5.0" and python_version >= "3.6"
dparse==0.5.1; python_version >= "3.5"
filelock==3.4.0; python_version >= "3.6" and python_full_version >= "3.6.1"
flake8-bandit==2.1.2
flake8-bugbear==21.9.2; python_version >= "3.6"
flake8-docstrings==1.6.0
flake8-polyfill==1.0.2
flake8-rst-docstrings==0.2.3; python_version >= "3.3"
flake8==4.0.1; python_version >= "3.6"
furo==2021.11.23; python_version >= "3.6"
gitdb==4.0.9; python_version >= "3.7"
gitpython==3.1.24; python_version >= "3.7"
identify==2.4.0; python_full_version >= "3.6.1"
idna==3.3; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.6.0" and python_version >= "3.6"
imagesize==1.3.0; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.4.0" and python_version >= "3.6"
importlib-metadata==4.2.0; python_version < "3.8" and python_version >= "3.6" and python_full_version >= "3.6.2"
iniconfig==1.1.1; python_version >= "3.6"
jinja2==3.0.3; python_version >= "3.6"
livereload==2.6.3; python_version >= "3.6"
markupsafe==2.0.1; python_version >= "3.6"
mccabe==0.6.1; python_version >= "3.6"
mypy-extensions==0.4.3; python_version >= "3.5" and python_full_version >= "3.6.2"
mypy==0.910; python_version >= "3.5"
nodeenv==1.6.0; python_full_version >= "3.6.1"
packaging==21.3; python_version >= "3.6"
pathspec==0.9.0; python_full_version >= "3.6.2"
pbr==5.7.0; python_version >= "3.6"
pep8-naming==0.12.1
platformdirs==2.4.0; python_version >= "3.6" and python_full_version >= "3.6.2"
pluggy==1.0.0; python_version >= "3.6"
pre-commit-hooks==4.0.1; python_full_version >= "3.6.1"
pre-commit==2.15.0; python_full_version >= "3.6.1"
py==1.11.0; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.5.0" and python_version >= "3.6"
pycodestyle==2.8.0; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.5.0" and python_version >= "3.6"
pydocstyle==6.1.1; python_version >= "3.6"
pyflakes==2.4.0; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.4.0" and python_version >= "3.6"
pygments==2.10.0; python_version >= "3.5"
pyparsing==3.0.6; python_version >= "3.6"
pytest==6.2.5; python_version >= "3.6"
pytz==2021.3; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.4.0" and python_version >= "3.6"
pyupgrade==2.29.1; python_full_version >= "3.6.1"
pyyaml==6.0; python_version >= "3.6" and python_full_version >= "3.6.1"
regex==2021.11.10; python_full_version >= "3.6.2"
reorder-python-imports==2.6.0; python_full_version >= "3.6.1"
requests==2.26.0; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.6.0" and python_version >= "3.6"
restructuredtext-lint==1.3.2; python_version >= "3.3"
ruamel.yaml.clib==0.2.6; platform_python_implementation == "CPython" and python_version < "3.10" and python_version >= "3.5" and python_full_version >= "3.6.1"
ruamel.yaml==0.17.17; python_version >= "3" and python_full_version >= "3.6.1"
safety==1.10.3; python_version >= "3.5"
six==1.16.0; python_full_version >= "3.6.1" and python_version >= "3.6"
smmap==5.0.0; python_version >= "3.7"
snowballstemmer==2.2.0; python_version >= "3.6"
soupsieve==2.3.1; python_full_version > "3.0.0" and python_version >= "3.6"
sphinx-autobuild==2021.3.14; python_version >= "3.6"
sphinx-click==3.0.2; python_version >= "3.6"
sphinx==4.3.0; python_version >= "3.6"
sphinxcontrib-applehelp==1.0.2; python_version >= "3.6"
sphinxcontrib-devhelp==1.0.2; python_version >= "3.6"
sphinxcontrib-htmlhelp==2.0.0; python_version >= "3.6"
sphinxcontrib-jsmath==1.0.1; python_version >= "3.6"
sphinxcontrib-qthelp==1.0.3; python_version >= "3.6"
sphinxcontrib-serializinghtml==1.1.5; python_version >= "3.6"
stevedore==3.5.0; python_version >= "3.6"
tokenize-rt==4.2.1; python_full_version >= "3.6.1"
toml==0.10.2; python_full_version >= "3.6.1" and python_version >= "3.6" and (python_version >= "3.5" and python_full_version < "3.0.0" or python_full_version >= "3.3.0" and python_version >= "3.5")
tomli==1.2.2; python_version >= "3.6" and python_full_version >= "3.6.2"
tornado==6.1; python_version >= "3.6"
typed-ast==1.4.3; python_version < "3.8" and python_version >= "3.5" and implementation_name == "cpython" and python_full_version >= "3.6.2"
typeguard==2.13.2; python_full_version >= "3.5.3"
typing-extensions==4.0.0
urllib3==1.26.7; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.6.0" and python_version < "4" and python_version >= "3.6"
virtualenv==20.10.0; python_full_version >= "3.6.1"
xdoctest==0.15.10
zipp==3.6.0; python_version < "3.8" and python_version >= "3.6" and python_full_version >= "3.6.1"

nox > python -m pip install --constraint=.nox/pre-commit/tmp/requirements.txt black darglint flake8 flake8-bandit flake8-bugbear flake8-docstrings flake8-rst-docstrings pep8-naming pre-commit pre-commit-hooks pyupgrade reorder-python-imports
Collecting black
  Using cached black-21.11b1-py3-none-any.whl (155 kB)
Collecting darglint
  Using cached darglint-1.8.1-py3-none-any.whl (120 kB)
Collecting flake8
  Using cached flake8-4.0.1-py2.py3-none-any.whl (64 kB)
Collecting flake8-bandit
  Using cached flake8_bandit-2.1.2-py3-none-any.whl
Collecting flake8-bugbear
  Using cached flake8_bugbear-21.9.2-py36.py37.py38-none-any.whl (16 kB)
Collecting flake8-docstrings
  Using cached flake8_docstrings-1.6.0-py2.py3-none-any.whl (5.7 kB)
Collecting flake8-rst-docstrings
  Using cached flake8_rst_docstrings-0.2.3-py3-none-any.whl (9.8 kB)
Collecting pep8-naming
  Using cached pep8_naming-0.12.1-py2.py3-none-any.whl (8.9 kB)
Collecting pre-commit
  Using cached pre_commit-2.15.0-py2.py3-none-any.whl (191 kB)
Collecting pre-commit-hooks
  Using cached pre_commit_hooks-4.0.1-py2.py3-none-any.whl (39 kB)
Collecting pyupgrade
  Using cached pyupgrade-2.29.1-py2.py3-none-any.whl (53 kB)
Collecting reorder-python-imports
  Using cached reorder_python_imports-2.6.0-py2.py3-none-any.whl (12 kB)
Collecting mypy-extensions>=0.4.3
  Using cached mypy_extensions-0.4.3-py2.py3-none-any.whl (4.5 kB)
Collecting regex>=2021.4.4
  Using cached regex-2021.11.10-cp310-cp310-macosx_10_9_x86_64.whl (288 kB)
Collecting pathspec<1,>=0.9.0
  Using cached pathspec-0.9.0-py2.py3-none-any.whl (31 kB)
Collecting click>=7.1.2
  Using cached click-8.0.3-py3-none-any.whl (97 kB)
Collecting typing-extensions>=3.10.0.0
  Using cached typing_extensions-4.0.0-py3-none-any.whl (22 kB)
Collecting tomli<2.0.0,>=0.2.6
  Using cached tomli-1.2.2-py3-none-any.whl (12 kB)
Requirement already satisfied: platformdirs>=2 in ./.nox/pre-commit/lib/python3.10/site-packages (from black) (2.4.0)
Collecting pyflakes<2.5.0,>=2.4.0
  Using cached pyflakes-2.4.0-py2.py3-none-any.whl (69 kB)
Collecting mccabe<0.7.0,>=0.6.0
  Using cached mccabe-0.6.1-py2.py3-none-any.whl (8.6 kB)
Collecting pycodestyle<2.9.0,>=2.8.0
  Using cached pycodestyle-2.8.0-py2.py3-none-any.whl (42 kB)
Collecting bandit
  Using cached bandit-1.7.1-py3-none-any.whl (113 kB)
Collecting flake8-polyfill
  Using cached flake8_polyfill-1.0.2-py2.py3-none-any.whl (7.3 kB)
Collecting attrs>=19.2.0
  Using cached attrs-21.2.0-py2.py3-none-any.whl (53 kB)
Collecting pydocstyle>=2.1
  Using cached pydocstyle-6.1.1-py3-none-any.whl (37 kB)
Collecting pygments
  Using cached Pygments-2.10.0-py3-none-any.whl (1.0 MB)
Collecting restructuredtext-lint
  Using cached restructuredtext_lint-1.3.2-py3-none-any.whl
Collecting pyyaml>=5.1
  Using cached PyYAML-6.0-cp310-cp310-macosx_10_9_x86_64.whl (197 kB)
Collecting nodeenv>=0.11.1
  Using cached nodeenv-1.6.0-py2.py3-none-any.whl (21 kB)
Collecting toml
  Using cached toml-0.10.2-py2.py3-none-any.whl (16 kB)
Requirement already satisfied: virtualenv>=20.0.8 in ./.nox/pre-commit/lib/python3.10/site-packages (from pre-commit) (20.10.0)
Collecting identify>=1.0.0
  Using cached identify-2.4.0-py2.py3-none-any.whl (98 kB)
Collecting cfgv>=2.0.0
  Using cached cfgv-3.3.1-py2.py3-none-any.whl (7.3 kB)
Collecting ruamel.yaml>=0.15
  Using cached ruamel.yaml-0.17.17-py3-none-any.whl (109 kB)
Collecting tokenize-rt>=3.2.0
  Using cached tokenize_rt-4.2.1-py2.py3-none-any.whl (6.1 kB)
Collecting aspy.refactor-imports>=2.1.0
  Using cached aspy.refactor_imports-2.2.0-py2.py3-none-any.whl (8.0 kB)
Collecting cached-property
  Using cached cached_property-1.5.2-py2.py3-none-any.whl (7.6 kB)
Collecting snowballstemmer
  Using cached snowballstemmer-2.2.0-py2.py3-none-any.whl (93 kB)
Requirement already satisfied: six<2,>=1.9.0 in ./.nox/pre-commit/lib/python3.10/site-packages (from virtualenv>=20.0.8->pre-commit) (1.16.0)
Requirement already satisfied: backports.entry-points-selectable>=1.0.4 in ./.nox/pre-commit/lib/python3.10/site-packages (from virtualenv>=20.0.8->pre-commit) (1.1.1)
Requirement already satisfied: filelock<4,>=3.2 in ./.nox/pre-commit/lib/python3.10/site-packages (from virtualenv>=20.0.8->pre-commit) (3.4.0)
Collecting distlib<1,>=0.3.1
  Using cached distlib-0.3.3-py2.py3-none-any.whl (496 kB)
Collecting GitPython>=1.0.1
  Using cached GitPython-3.1.24-py3-none-any.whl (180 kB)
Collecting stevedore>=1.20.0
  Using cached stevedore-3.5.0-py3-none-any.whl (49 kB)
Collecting docutils<1.0,>=0.11
  Using cached docutils-0.17.1-py2.py3-none-any.whl (575 kB)
Collecting gitdb<5,>=4.0.1
  Using cached gitdb-4.0.9-py3-none-any.whl (63 kB)
Collecting pbr!=2.1.0,>=2.0.0
  Using cached pbr-5.7.0-py2.py3-none-any.whl (112 kB)
Collecting smmap<6,>=3.0.1
  Using cached smmap-5.0.0-py3-none-any.whl (24 kB)
Installing collected packages: smmap, pyflakes, pycodestyle, pbr, mccabe, gitdb, stevedore, snowballstemmer, pyyaml, GitPython, flake8, docutils, distlib, cached-property, typing-extensions, tomli, toml, tokenize-rt, ruamel.yaml, restructuredtext-lint, regex, pygments, pydocstyle, pathspec, nodeenv, mypy-extensions, identify, flake8-polyfill, click, cfgv, bandit, attrs, aspy.refactor-imports, reorder-python-imports, pyupgrade, pre-commit-hooks, pre-commit, pep8-naming, flake8-rst-docstrings, flake8-docstrings, flake8-bugbear, flake8-bandit, darglint, black
  Attempting uninstall: distlib
    Found existing installation: distlib 0.3.4
    Uninstalling distlib-0.3.4:
      Successfully uninstalled distlib-0.3.4
Successfully installed GitPython-3.1.24 aspy.refactor-imports-2.2.0 attrs-21.2.0 bandit-1.7.1 black-21.11b1 cached-property-1.5.2 cfgv-3.3.1 click-8.0.3 darglint-1.8.1 distlib-0.3.3 docutils-0.17.1 flake8-4.0.1 flake8-bandit-2.1.2 flake8-bugbear-21.9.2 flake8-docstrings-1.6.0 flake8-polyfill-1.0.2 flake8-rst-docstrings-0.2.3 gitdb-4.0.9 identify-2.4.0 mccabe-0.6.1 mypy-extensions-0.4.3 nodeenv-1.6.0 pathspec-0.9.0 pbr-5.7.0 pep8-naming-0.12.1 pre-commit-2.15.0 pre-commit-hooks-4.0.1 pycodestyle-2.8.0 pydocstyle-6.1.1 pyflakes-2.4.0 pygments-2.10.0 pyupgrade-2.29.1 pyyaml-6.0 regex-2021.11.10 reorder-python-imports-2.6.0 restructuredtext-lint-1.3.2 ruamel.yaml-0.17.17 smmap-5.0.0 snowballstemmer-2.2.0 stevedore-3.5.0 tokenize-rt-4.2.1 toml-0.10.2 tomli-1.2.2 typing-extensions-4.0.0

nox > pre-commit run --all-files --show-diff-on-failure
black....................................................................Passed
Check for added large files..............................................Passed
Check Toml...............................................................Passed
Check Yaml...............................................................Passed
Fix End of Files.........................................................Passed
flake8...................................................................Passed
pyupgrade................................................................Passed
Reorder python imports...................................................Passed
Trim Trailing Whitespace.................................................Passed
prettier.................................................................Passed
nox > Session pre-commit was successful.
```

But of course if I blow away that virtual environment and regenerated the error will re-appear again.

Anyone have suggestions on how to work around this issue?

Thanks so much.

Cheers

Doug

**Comment by dlethin** [comment]
So I think I figured out where I strayed.  I mentioned that I was "following the quickstart guide...", but it turns out I wasn't following it verbatim.    I  took a different path when installing poetry, choosing to use the installer mentioned in the poetry docs which I've now learned is deprecated,  rather than use the install-poetry.py method which apparently is the new method going forward.

Turns out the old install method is the one with the missing dependencies.    After uninstalling poetry and re-installing via the preferred new approach,  I was able to eliminate the error I was getting running nox.

I'm going to close this issue.  Thanks for what looks like I a great python starter project template.

Cheers.
